### ISSUE TITLE
New Feature: parse and reformat date in view

### ISSUE THREAD

**Issue by bordaigorl** [FichteFoll/InsertDate#29]
It would be nice to have a command to parse the selection in a view assuming it's a date, then present the user with a panel with the output formats to which the date can be transformed.

**Comment by FichteFoll** [comment]
That's a good idea. I will couple that with #15 when I work on it.

How would you suggest to determine the format of the timestamp parsed?

**Comment by bordaigorl** [comment]
What about using a quickpanel just as in `InsertDatePanelCommand`? For me that would be ideal.
There could also be the option to specify a fixed format as an argument of a dedicated command.

**Comment by FichteFoll** [comment]
Yeah, I could just use the same panel as in `InsertDatePanelCommand`. I could also attempt to parse the string(s) with all format strings from the `prompt_config` setting (minus tz_in and tz_out variants) and take whatever matches - and only ask for an explicit format if none matches.

If the above works like I think it could, i'll most likely implement it like that.

**Comment by bordaigorl** [comment]
If I were to do it, I would try to see if the `dateutil` works with ST's python (it is not bundled with ST3 as far as I can see).
It contains a `parse` function that tries to guess the format (see [here](https://labix.org/python-dateutil#head-c0e81a473b647dfa787dc11e8c69557ec2c3ecd2)).
As an ugly fallback, if nothing else works, you could try parsing the date using each template with `datetime.strptime`.

Also, I would probably not write new command but exploit the current ones adding a, say, `date` optional argument (default to `None`): when `None` do as now (backward compat), when `now` or `tomorrow` or few other keywords compute the corresponding date, when `selection` or a generic string try to parse.

**Comment by FichteFoll** [comment]
Thanks for the hint. `dateutil` looks like it could easily be made compatible with ST and I can just bundle it as a dependency, like I did with `pytz`.


I think that would be enough infos to start working on this.
Don't expect it too soon though, I have many things with higher priority currently, so for the moment you'll either have to stick to regular expressions or implement it on your on and file a pull request.
